src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesBackupProtectionPolicy.cs

```diff
@@ -32,28 +32,42 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
     [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesBackupProtectionPolicy",SupportsShouldProcess = true), OutputType(typeof(JobBase))]
     public class SetAzureRmRecoveryServicesBackupProtectionPolicy : RSBackupVaultCmdletBase
     {
+        public const string ModifyPolicyParamSet = "ModifyPolicyParamSet";
+        public const string FixInconsistentPolicyParamSet = "FixPolicyParamSet";
         /// <summary>
         /// Policy object to be modified
         /// </summary>
         [Parameter(Position = 1, Mandatory = true, HelpMessage = ParamHelpMsgs.Policy.ProtectionPolicy,
-            ValueFromPipeline = true)]
+            ValueFromPipeline = true, ParameterSetName = ModifyPolicyParamSet)]
+        [Parameter(Position = 1, Mandatory = true, HelpMessage = ParamHelpMsgs.Policy.ProtectionPolicy,
+            ValueFromPipeline = true, ParameterSetName = FixInconsistentPolicyParamSet)]
         [ValidateNotNullOrEmpty]
         public PolicyBase Policy { get; set; }
 
         /// <summary>
         /// Retention policy object to be modified
         /// </summary>
-        [Parameter(Position = 2, Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.RetentionPolicy)]
+        [Parameter(Position = 2, Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.RetentionPolicy,
+            ParameterSetName = ModifyPolicyParamSet)]
         [ValidateNotNullOrEmpty]
         public RetentionPolicyBase RetentionPolicy { get; set; }
 
         /// <summary>
         /// Schedule policy object to be modified
         /// </summary>
-        [Parameter(Position = 3, Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.SchedulePolicy)]
+        [Parameter(Position = 3, Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.SchedulePolicy,
+            ParameterSetName = ModifyPolicyParamSet)]
         [ValidateNotNullOrEmpty]
         public SchedulePolicyBase SchedulePolicy { get; set; }
 
+        /// <summary>
+        /// Retry Policy Update for Failed Items
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.FixForInConsistentItems,
```

Comment:
I think this switch parameter **is** mandatory in this parameter set, right?

Reply:
And `ValidateNotNullOrEmpty` won't be necessary

Reply:
Besides, since you are changing the cmdlet surface, please regenerate help document.

Reply:
Yes, will do the changes